src/materialized/mux.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use async_trait::async_trait;
+use futures::future::TryFutureExt;
+use log::error;
+use tokio::io::{AsyncRead, AsyncWrite, AsyncWriteExt};
+
+use ore::netio::{self, SniffedStream, SniffingStream};
+
+use crate::http;
+
+pub struct Mux<A> {
```

Comment:
Could you add a docco explaining what this does?

Reply:
I guess I mean its primary responsibilities, looking at the code it's obvious what kind of mux it is, but we have lots of multiplexers with a lot of responsibilities.

Reply:
You bet, done.